pkg/credentials/kubernetes/acr/workload_identity.go

```diff
@@ -0,0 +1,199 @@
+package acr
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/containers/azcontainerregistry"
+	"github.com/patrickmn/go-cache"
+
+	"github.com/akuity/kargo/pkg/credentials"
+	"github.com/akuity/kargo/pkg/logging"
+)
+
+const (
+	// ACRTokenValidityHours is the typical validity period for ACR refresh tokens
+	ACRTokenValidityHours = 3
+	// CacheTTLMinutes is how long we cache ACR tokens before refreshing them.
+	// Set to 2.5 hours to ensure we refresh before the 3-hour token expiry.
+	CacheTTLMinutes = 150
+	// CleanupIntervalMinutes is how often the cache cleanup runs
+	CleanupIntervalMinutes = 30
+	// ACRTokenUsername is the fixed username used for ACR token authentication
+	ACRTokenUsername = "00000000-0000-0000-0000-000000000000"
+	// ACRScope is the Azure AD scope required for ACR authentication
+	ACRScope = "https://containerregistry.azure.net/.default"
+)
+
+// WorkloadIdentityProvider implements credentials.Provider for Azure Container Registry
+// workload identity authentication.
+type WorkloadIdentityProvider struct {
+	tokenCache *cache.Cache
+	credential azcore.TokenCredential
+
+	getAccessTokenFn func(ctx context.Context, registryName string) (string, error)
+}
+
+// NewWorkloadIdentityProvider returns a new WorkloadIdentityProvider
+// if Azure workload identity credentials are available.
+func NewWorkloadIdentityProvider(ctx context.Context) credentials.Provider {
+	logger := logging.LoggerFromContext(ctx)
+
+	// Try to create a DefaultAzureCredential which supports workload identity
+	credential, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		logger.Info("Azure workload identity not available", "error", err.Error())
+		return nil
+	}
+
+	logger.Info("Azure workload identity credential provider initialized")
+
+	p := &WorkloadIdentityProvider{
+		tokenCache: cache.New(
+			CacheTTLMinutes*time.Minute,        // Default ttl for each entry
+			CleanupIntervalMinutes*time.Minute, // Cleanup interval
+		),
+		credential: credential,
+	}
+	p.getAccessTokenFn = p.getAccessToken
+	return p
+}
+
+func (p *WorkloadIdentityProvider) Supports(
+	credType credentials.Type,
+	repoURL string,
+	_ map[string][]byte,
+	_ map[string]string,
+) bool {
+	if p.credential == nil {
+		return false
+	}
+
+	if credType != credentials.TypeImage && credType != credentials.TypeHelm {
+		return false
+	}
+
+	if credType == credentials.TypeHelm && !strings.HasPrefix(repoURL, "oci://") {
+		return false
+	}
```

Comment:
This needs to be removed because, since #4995, repoURLs are normalized before ever being passed to the providers and normalization for an OCI chart repo URL will have removed the leading `oci://`.